### ISSUE TITLE
QUESTION: Is it possible to specify `cross_fields` in DataSearch generated ES Query?

### ISSUE THREAD

**Issue by tankchintan** [appbaseio/reactivesearch#1142]
Hello I have recently started using your excellent project & I fell it provides lots of extension points.

I was wondering though is it possible to also do a multi_match using `cross_fields`. I know you are using `best_fields` & `phrase_prefix` right now.

I know I could possibly use `customQuery` to accomplish this but I am afraid that I might end up duplicating lot of logic you have already baked in, in the standard e.g. using field weights, fuzziness etc.

Alternatively if there is a way for me to use a method which generates the query... in my customQuery implementation, that should also be fine. 

Here is the context on why I would need cross_fields as well.

I am using ElasticSearch 7.3. I am making a query against multiple fields using an "universal" search box. E.g. I intend users to just type in a search query without trying to fill in specific search boxes for specific fields.

On my part to reduce complexity I search across few selected fields only for this use case. Let's say name, address.

```
  // Target of copy_to of multiple fields
  address:
    type: text  

  // Target of copy_to of multiple fields
  name:
    type: text
```

With this I would expect to get the first hit to be a document with,

```
name: chantz lasts
address 9090 auburn drive KY
```

But instead I see stuff like,
```
name: chantz hasts
address 1122 castllight drive KY

name: chantz blasts
address 5656 suma pl CA

```

Basically no results at least in the first 10 hits that should match terms present in both the fields. i.e. 
```
name: chantz lasts
address 9090 auburn drive KY
```

If I switch the order of the query from `chantz auburn` to `auburn chantz` then I get similar results but this time with address having `auburn` in them but the name does not have `chantz` in it.

I then tried `cross_fields` by using the query geenrated from reactivesearch & it worked.
